ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -175,8 +175,16 @@ func convertToContainerDef(inputCfg *adapter.ContainerConfig, ecsContainerDef *C
 		mem = resolveIntResourceOverride(inputCfg.Name, mem, ecsMemInMB, "MemoryLimit")
 
 		ecsMemResInMB := adapter.ConvertToMemoryInMB(int64(ecsContainerDef.MemoryReservation))
+
 		memRes = resolveIntResourceOverride(inputCfg.Name, memRes, ecsMemResInMB, "MemoryReservation")
 
+		credParam := ecsContainerDef.RepositoryCredentials.CredentialsParameter
+
+		if credParam != "" {
+			outputContDef.RepositoryCredentials = &ecs.RepositoryCredentials{}
```

Comment:
This it totally fine, but just wondering aloud: We are sometimes inconsistent with using the setters on objects vs just assigning the field to an empty version of it. E.g. we're using `SetCredentialsParameter` in line 185, but not `SetRepositoryCredentials` in line 184. Is this something we might want to think about?

Reply:
True, we are inconsistent with this. But they're each more concise depending on what you want to do.

In the case of setting up a `RepositoryCredentials`, if I wanted to use the setter method I'd have to create an empty struct first, and then set it with the method (otherwise it'd set it to an empty pointer, so I wouldn't have a thing to call `SetCredentialsParameter` on), so setting directly like on line 184 just lets me do it in one line. On 185 the value I want to set already exists, so I use the setter method, which also takes care of converting it to a pointer so I don't have to do `aws.String()`.

This isn't definitive guidance of course, but perhaps using either depending on what results in fewer lines of code might be one way of deciding?